pkg/controller/jobframework/reconciler.go

```diff
@@ -381,22 +381,26 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 
 	jobPodSets := job.PodSets()
 
-	if !workload.CanBePartiallyAdmitted(wl) || job.IsSuspended() {
+	if !workload.CanBePartiallyAdmitted(wl) || !workload.IsAdmitted(wl) {
```

Comment:
Why this? I think we can end up recreating the workload when ever it switches off from admitted (is evicted)

Reply:
when a workload is evicted, it still counts as admitted.
We only clear the status.admission once the job is suspended

https://github.com/kubernetes-sigs/kueue/blob/6faebaffa5d1c856a6a0f56423d0b665dbd27fe6/pkg/controller/jobframework/reconciler.go#L219-L233